subprojects/core/src/main/java/org/gradle/api/tasks/WriteProperties.java

```diff
@@ -191,8 +195,12 @@ public void setEncoding(String encoding) {
      * Returns the output file to write the properties to.
      */
     @OutputFile
+    @Optional
     public File getOutputFile() {
-        return getServices().get(FileOperations.class).file(outputFile);
+        if(outputFile != null) {
+            return getServices().get(FileOperations.class).file(outputFile);
+        }
+        return null;
```

Comment:
:x: This introduce a disconnected new property. You should instead have the old one delegates to the new one. e.g.
```suggestion
        return destinationFile.getAsFile().getOrNull();
```

and have the setter for `outputFile` actually call `destinationFile.set(...)`.